Gateway/Http/Client/TransactionRefund.php

```diff
@@ -55,9 +55,9 @@ public function __construct(
     public function placeRequest(TransferInterface $transferObject): array
     {
         $requests = $transferObject->getBody();
-        $headers = $transferObject->getHeaders();
-        $clientConfig = $transferObject->getClientConfig();
+        $requestOptions['headers'] = $this->requestHeaders($transferObject);
```

Comment:
Ideally `$transferObject->getHeaders()` should be sufficient to fetch the headers of the requests. Implementing an abstract class for an abstract method would increase the complexity of the design. Let's discuss this approach.